verifier/verifier.go

```diff
@@ -59,7 +59,19 @@ func Verify(layout *Layout, attestations map[string]*dsse.Envelope, parameters m
 
 		acceptedKeys, err := envVerifier.Verify(context.Background(), env)
 		if err != nil {
-			return err
+			// The verifier loads all attestations and verifies their
+			// signatures. It represents their claims in the format "<signer>
+			// says <claim> for <step>", allowing policy to be written as "does
+			// one of <trusted signers> say <claim> for <step>?"
+			// While this might result in verifying the signatures of
+			// attestations that aren't required for the specific layout, it
+			// more cleanly separates policy evaluation from claim expression.
+			// Also, we do not authenticate attestations from unknown verifiers,
+			// we only hit a situation here where trusted keys from a trusted
+			// layout cannot authenticate an attestation, which is discarded and
+			// not used in the rest of the verification workflow.
+			log.Infof("Unable to verify %s's signatures", attestationName)
+			continue
```

Comment:
How is it discarded? That's not obvious to me right now.

Reply:
updated text